bootstore/src/schemes/v0/state.rs

```diff
@@ -0,0 +1,594 @@
+    },
+    /// We only store the computed secret in memory for so long after it has
+    /// been computed
+    Computed { secret: RackSecret, expiry: Ticks },
+}
+
+impl RackSecretState {
+    pub fn load(
+        &mut self,
+        rack_uuid: Uuid,
+        peers: &BTreeSet<Baseboard>,
+        local_id: &Baseboard,
+        local_share: &Vec<u8>,
+        new_expiry: Ticks,
+        threshold: usize,
+        share_digests: &Vec<Sha3_256Digest>,
+    ) -> Output {
+        match self {
+            RackSecretState::Empty => {
+                // Start the share retrieval process
+                let (request_id, output) =
+                    self.broadcast_share_requests(rack_uuid, peers);
+                *self = RackSecretState::Retrieving {
+                    request_id,
+                    from: BTreeSet::from([local_id.clone()]),
+                    shares: Shares(vec![local_share.clone()]),
+                    expiry: new_expiry,
+                    threshold,
+                    share_digests: share_digests.clone(),
+                };
+                output
+            }
+            RackSecretState::Retrieving { expiry, .. } => {
+                // We already have a retrieval in progress.
+                // Just extend the expiry
+                *expiry = new_expiry;
+                Output::none()
+            }
+            RackSecretState::Computed { secret, .. } => {
+                // We have the secret, so return it.
+                Output {
+                    persist: false,
+                    envelopes: vec![],
+                    api_output: Some(Ok(ApiOutput::RackSecret(secret.clone()))),
+                }
+            }
+        }
+    }
+
+    // Did the share retrieval process timeout ?
+    pub fn on_tick(&mut self, now: Ticks) -> Output {
+        match self {
+            RackSecretState::Retrieving { expiry, .. } => {
+                if *expiry < now {
+                    *self = RackSecretState::Empty;
+                    return ApiError::RackSecretLoadTimeout.into();
+                }
+            }
+            RackSecretState::Computed { expiry, .. } => {
+                if *expiry < now {
+                    // This is just cleanup of the secret from memory
+                    // No reason to report this
+                    *self = RackSecretState::Empty;
+                }
+            }
+            RackSecretState::Empty => (),
+        }
+        Output::none()
+    }
+
+    fn validate_share(
+        from: &Baseboard,
+        share: &Vec<u8>,
+        share_digests: &Vec<Sha3_256Digest>,
+    ) -> Result<(), ApiError> {
+        let computed_hash = Sha3_256Digest(
+            Sha3_256::digest(share).as_slice().try_into().unwrap(),
+        );
+
+        if !share_digests.contains(&computed_hash) {
+            Err(ApiError::InvalidShare { from: from.clone() })
+        } else {
+            Ok(())
+        }
+    }
+
+    // Send a request for a share to each peer
```

Comment:
This docstring is misleading; it's generating an `Output` and `Uuid`, but is itself not doing any sending

Reply:
No longer relevant as of [98ada27](https://github.com/oxidecomputer/omicron/pull/3385/commits/98ada27b53d4429463cde94cc91bedec0bc4ef98)